Source/Redistribution/iamr_create_itracker_2d.cpp

```diff
@@ -0,0 +1,364 @@
+           const Real dapy = apy(i  ,j+1,k  ) - apy(i,j,k);
+           apnorm = std::sqrt(dapx*dapx+dapy*dapy);
+           apnorm_inv = 1.0/apnorm;
+           Real nx = dapx * apnorm_inv;
+           Real ny = dapy * apnorm_inv;
+
+           // We use small_norm as an offset just to break the tie when at 45 degrees ...
+
+           if (preferred_direction == 1)
+           {
+               // y-direction is preferred
+               if (nx > 0)
+                  nx -= small_norm;
+               else
+                  nx += small_norm;
+
+           } else {
+               // x-direction is preferred
+               if (ny > 0)
+                  ny -= small_norm;
+               else
+                  ny += small_norm;
+           }
+
+           // As a first pass, choose just based on the normal
+           if (std::abs(nx) > std::abs(ny))
+           {
+               if (nx > 0)
+                   itracker(i,j,k,1) = 5;
+               else
+                   itracker(i,j,k,1) = 4;
+
+           } else {
+               if (ny > 0)
+                   itracker(i,j,k,1) = 7;
+               else
+                   itracker(i,j,k,1) = 2;
+           }
+
+           bool xdir_mns_ok = (is_periodic_x || (i != domain.smallEnd(0)));
+           bool xdir_pls_ok = (is_periodic_x || (i != domain.bigEnd(0)  ));
+           bool ydir_mns_ok = (is_periodic_y || (j != domain.smallEnd(1)));
+           bool ydir_pls_ok = (is_periodic_y || (j != domain.bigEnd(1)  ));
+
+           // Override above logic if trying to reach outside a domain boundary (and non-periodic)
+           if ( (!xdir_mns_ok && (itracker(i,j,k,1) == 4)) ||
+                (!xdir_pls_ok && (itracker(i,j,k,1) == 5)) )
+           {
+               itracker(i,j,k,1) = (ny > 0) ? 7 : 2;
+           }
+           if ( (!ydir_mns_ok && (itracker(i,j,k,1) == 2)) ||
+                (!ydir_pls_ok && (itracker(i,j,k,1) == 7)) )
+           {
+               itracker(i,j,k,1) = (nx > 0) ? 5 : 4;
+           }
+
+           // (i,j) merges with at least one cell now
+           itracker(i,j,k,0) += 1;
+
+           // (i+ioff,j+joff) is in the nbhd of (i,j)
+           int ioff = imap[itracker(i,j,k,1)];
+           int joff = jmap[itracker(i,j,k,1)];
+
+           // Sanity check
+           if (vfrac(i+ioff,j+joff,k) == 0.)
+               amrex::Abort(" Trying to merge with covered cell");
+
+           Real sum_vol = vfrac(i,j,k) + vfrac(i+ioff,j+joff,k);
+
+#if 0
+           if (debug_verbose > 0)
+               amrex::Print() << "Cell " << IntVect(i,j) << " with volfrac " << vfrac(i,j,k) <<
+                                 " trying to merge with " << IntVect(i+ioff,j+joff) <<
+                                 " with volfrac " << vfrac(i+ioff,j+joff,k) <<
+                                 " to get new sum_vol " <<  sum_vol << std::endl;
+#endif
+
+           // If the merged cell isn't large enough, we try to merge in the other direction
+           if (sum_vol < 0.5)
```

Comment:
Will this ever be a parameter that will require tuning, or is this a magic number?

Reply:
I don't think this will require tuning